File: Day12/Solution.java
class Student extends Person{
	private int[] testScores;

    public Student(String firstname,String lastname,int id,int[] Scores)
    {
            super(firstname,lastname,id);
            this.testScores =Scores;
    }

    public char calculate(){
        double avg = 0.0;
        for(int i=0;i<testScores.length;i++)
            avg=avg + testScores[i];
        avg=avg/testScores.length;
        if(avg<40)
            return 'T';
        else if(avg<55)
            return 'D';
            else if (avg<70)
                    return 'P';
                else if(avg<80)
                    return 'A';
                    else if(avg<90)
                        return 'E';
                        else
                            return 'O';
    }

}

File: Day12/Solution.py
class Student(Person):
    def __init__(self, fName, lName, sId, scores):
        super().__init__(fName, lName, sId)
        self.scores = scores

    def calculate(self):
        avg = 0.0
        for score in self.scores:
            avg += score

        avg = avg/len(self.scores)
        if avg < 40:
            return 'T'
        elif avg < 55:
            return 'D'
        elif avg < 70:
            return 'P'
        elif avg < 80:
            return 'A'
        elif avg < 90:
            return 'E'
        else:
            return 'O'
